aeternity/oracle.py

```diff
@@ -0,0 +1,348 @@
+import json
+from collections import defaultdict, namedtuple
+
+import logging
+from json import JSONDecodeError
+
+import requests
+import time
+import websocket
+
+from aeternity.config import Config
+
+logger = logging.getLogger(__name__)
+
+
+class Connection:
+    def __init__(self, config):
+        self.config = config
+        self.websocket = None
+
+    def close(self):
+        self.websocket.close()
+        self.websocket = None
+
+    def assure_connected(self):
+        if self.websocket is None:
+            self.websocket = websocket.create_connection(self.config.websocker_url)
+
+    def receive(self):
+        self.assure_connected()
+        message = json.loads(self.websocket.recv())
+        logger.debug('received: %s', message)
+        return message
+
+    def send(self, message):
+        self.assure_connected()
+        logger.debug('sending: %s', message)
+        self.websocket.send(json.dumps(message))
+
+
+class EpochRequestError(Exception):
+    pass
+
+
+class EpochClient:
```

Comment:
I think `EpochClient` should be pulled out of the oracles module.

Reply:
Yeah, I've just done that. It just grew to be like this and I noticed myself this morning